inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1131,6 +1131,55 @@ void FuseMultipleIdentitiesPass::run() {
     }
 }
 
+void BroadcastConstPass::run() {
+    for (auto& constLayer : *pLayers) {
+        if (!LayerInfo(constLayer).isConst()) {
+            continue;
+        }
+        auto isNonFunctional = [](CNNLayerPtr l) {
+            return LayerInfo(l).isNonFunctional();
+        };
+        if (!CNNNetHasNextLayerSkipCertain(constLayer, 0, 0, isNonFunctional)) {
+            continue;
+        }
+
+        auto nextLayer = CNNNetGetNextLayerSkipCertain(constLayer, 0, 0, isNonFunctional).first;
+
+        if (!LayerInfo(nextLayer).isEltwise()) {
+            continue;
+        }
+
+        auto constDims = constLayer->outData.front()->getTensorDesc().getDims();
+        auto constDimsSize = InferenceEngine::details::product(constDims.begin(), constDims.end());
+        auto eltwiseDims = nextLayer->outData.front()->getTensorDesc().getDims();
+        auto eltwiseDimsSize = InferenceEngine::details::product(eltwiseDims.begin(), eltwiseDims.end());
+
+        if (constDimsSize == eltwiseDimsSize) {
+            continue;
+        }
+        if (eltwiseDimsSize % constDimsSize != 0) {
+            continue;
+        }
+
+        auto currentConstBlob = constLayer->blobs.begin()->second;
+        auto constPrecision = constLayer->blobs.begin()->second->getTensorDesc().getPrecision();
+        auto newConstBlob = make_blob_with_precision(InferenceEngine::TensorDesc(constPrecision,
+            nextLayer->outData.front()->getTensorDesc().getDims(),
+            nextLayer->outData.front()->getTensorDesc().getLayout()));
+        newConstBlob->allocate();
+        std::size_t offset = 0;
+        for (std::size_t i = 0; i < (eltwiseDimsSize / constDimsSize); ++i) {
+            ie_memcpy(newConstBlob->buffer().as<char*>() + (i * currentConstBlob->byteSize()),
```

Comment:
please check possibility to use 
``` c++
auto tileBlob = [](Blob::Ptr &blob, size_t TileTo){
```

from substituteScaleshiftBroadcast

Reply:
tileBlob from substituteScaleshiftBroadcast assigns new blob to passed pointer. Same action doesn't work for this broadcast (output is incorrect if I do so) and in that case the new blob needs to be assigned directly to constLayer->blobs.find("custom")->second. I can modify the tileBlob to return a blob, not bool, so that each pass will assign the created blob how they need to.

Reply:
sure modify as needed just keep logic of tiling - so memcpy and loop over it should be moved to helper function - currently we keep helpers in gna_pass_manager.cpp file as static function